PocketSocket/PSWebSocket.m

```diff
@@ -160,14 +160,22 @@ - (instancetype)initClientSocketWithRequest:(NSURLRequest *)request {
         if(port == 0) {
             port = (_secure) ? 443 : 80;
         }
-        
+
         CFReadStreamRef readStream = nil;
         CFWriteStreamRef writeStream = nil;
         CFStreamCreatePairWithSocketToHost(kCFAllocatorDefault,
                                            (__bridge CFStringRef)host,
                                            port,
                                            &readStream,
                                            &writeStream);
+
+        CFDictionaryRef systemProxyConfig = CFNetworkCopySystemProxySettings();
```

Comment:
please bear in mind that I have 0 experience with objective-c. Any comments are very welcome! 🙇🏽

Reply:
@dgvncsz0f Given that changes to this submod will directly influence Sketch too, is there an associated ticket for this in Sketch?

Reply:
@InvisiblePixels https://github.com/sketch-hq/Cloud/issues/6520 would be it, but I could provide more context if you need to (or update the ticket with any missing information) . I've done some preliminary tests and this seems to work, but I'm expecting someone familiar with the codebase to sign off this, ofc, before merging it and applying this to sketch.

Reply:
But the problem is this: You want to merge a change into a submodule used by Sketch. Doing so will change the commit hash of this repository's develop branch's HEAD.

You will also need to create a ticket in Sketch in order to update the submod pointer contained in Sketch.

You can't just go changing submods without also updating Sketch.

Reply:
Just to follow up and clarify @dgvncsz0f the issue is we need a PR in the sketch repo to bring in the updated submod.  To have a PR in the sketch repo we need an associated ticket in the sketch repo.  Even if this gets merged it won't affect Sketch in any way until there's a Sketch PR to use the new pocketsocket :)

Reply:
ok, my mistake for not having a clear picture of the process, I apologize. 

I have this one https://github.com/sketch-hq/Sketch/pull/34578 but I thought I'd need this merged in order to have that one ready for review, I mean, the one that changes the submodule in Sketch. I'm aware that I need to update the submodule in sketch, but I thought I'd need a hash that points to master or develop (or any other release).

I was using that ticket to reference everything. This seems like a problem, if I understand you correctly. It should be a ticket in Sketch repo, correct?

Summarizing:

* create a ticket in sketch (could I create this myself);
* use that ticket as reference for PRs, and not a cloud one; 
* create a ticket in Sketch that updates the submodule pointing to this ref that is under review;
* have QA validate that this works;
* merge PocketSocket PR;
* update the PR in sketch, to point to the final hash;
* QA again (?);
* merge Sketch;

Would that be the correct procedure?

Reply:
You don't need to create two tickets in Sketch :)

1. Create a ticket in Sketch.
1. Create a feature branch in Sketch _and_ in the submod (the `bo` tools will do this for you) and make your changes in the feature branch.
1. Create a PR with your changes (`bo` will do this too and deal with the submod properly)
1. Pass the PR through the Code Review process.
1. Pass through QA.
1. Changes to submods require manual merging, so the PR must have the manual merge label applied.
1. Once approved notify me (or Mike or Pieter) to do the manual merge.

Devs never _ever_ manually merge stuff into Sketch. PRs either get merged by mergebot or manually by Pieter Mike or myself.

Reply:
👍🏽 thanks for clarifying and apologies again. I'm closing this one and open new ones that follows the correct procedure. Thanks in advance!